packages/core/src/node/cluster/main.ts

```diff
@@ -1,71 +0,0 @@
-/********************************************************************************
- * Copyright (C) 2017 TypeFox and others.
- *
- * This program and the accompanying materials are made available under the
- * terms of the Eclipse Public License v. 2.0 which is available at
- * http://www.eclipse.org/legal/epl-2.0.
- *
- * This Source Code may also be made available under the following Secondary
- * Licenses when the conditions for such availability set forth in the Eclipse
- * Public License v. 2.0 are satisfied: GNU General Public License, version 2
- * with the GNU Classpath Exception which is available at
- * https://www.gnu.org/software/classpath/license.html.
- *
- * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
- ********************************************************************************/
-
-import * as cluster from 'cluster';
-import { checkParentAlive } from '../messaging/ipc-protocol';
-import { MasterProcess } from './master-process';
-import { BackendApplicationConfigProvider } from '../backend-application-config-provider';
-
-checkParentAlive();
-
-process.on('unhandledRejection', (reason, promise) => {
-    throw reason;
-});
-
-import yargs = require('yargs');
-const args = yargs.option(MasterProcess.startupTimeoutOption, {
-    description: 'The number of milliseconds to wait for the server to start up. Pass a negative number to disable the timeout.',
-    type: 'number',
-    default: BackendApplicationConfigProvider.get().startupTimeout || MasterProcess.defaultStartupTimeoutOption
-}).help(false).argv;
-const noCluster = args['cluster'] === false;
-const isMaster = !noCluster && cluster.isMaster;
-const development = process.env.NODE_ENV === 'development';
-
-const startupTimeout = args[MasterProcess.startupTimeoutOption] as number;
```

Comment:
If the startup timeout is removed, please deprecate https://github.com/theia-ide/theia/blob/ce2f145017ff6bca715cddb96f876366eb1a4b22/dev-packages/application-package/src/application-props.ts#L124